hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -105,6 +108,25 @@ public boolean needsUpgradeOrDowngrade(HoodieTableVersion toVersion) {
    * @param instantTime current instant time that should not be touched.
    */
   public void run(HoodieTableVersion toVersion, String instantTime) {
+    // Change metadata table version automatically
+    if (toVersion.versionCode() >= HoodieTableVersion.FOUR.versionCode()) {
+      String metadataTablePath = HoodieTableMetadata.getMetadataTableBasePath(
+          metaClient.getBasePathV2().toString());
+      try {
+        if (metaClient.getFs().exists(new Path(metadataTablePath))) {
+          HoodieTableMetaClient mdtMetaClient = HoodieTableMetaClient.builder()
+              .setConf(metaClient.getHadoopConf()).setBasePath(metadataTablePath).build();
+          HoodieWriteConfig mdtWriteConfig = HoodieBackedTableMetadataWriter.createMetadataWriteConfig(
+              config, HoodieFailedWritesCleaningPolicy.LAZY);
```

Comment:
why this has to be lazy? we can rollback any pending operations right ?

Reply:
So, this takes care of both upgrade and downgrade right. i.e. upgrade will upgrade both data table and mdt table and downgrade will downgrade both ?

Reply:
Upgrade and downgrade do not do any write or cleaning so the `HoodieFailedWritesCleaningPolicy` is not used.  For consistency, yeah, I changed it to eager.

Reply:
Yes, the upgrade will also be taken care of.  However, that is not a usability issue before this PR, since when writing to a table, before the metadata table writer commits to the timeline, it automatically checks if the metadata table needs to be upgraded first.

Reply:
gotcha.